src/Nethermind/Nethermind.JsonRpc/Modules/Eth/EthRpcModule.cs

```diff
@@ -407,9 +407,11 @@ public ResultWrapper<BlockForRpc> eth_getBlockByNumber(BlockParameter blockParam
             return ResultWrapper<string?>.Success(null);
         }
 
-        IByteBuffer buffer = PooledByteBufferAllocator.Default.Buffer(TxDecoder.Instance.GetLength(transaction, RlpBehaviors.None));
+        RlpBehaviors encodingSettings = RlpBehaviors.SkipTypedWrapping | (transaction.IsInMempoolForm() ? RlpBehaviors.InMempoolForm : RlpBehaviors.None);
+
+        IByteBuffer buffer = PooledByteBufferAllocator.Default.Buffer(TxDecoder.Instance.GetLength(transaction, encodingSettings));
```

Comment:
btw return buffer?

`finally { byteBuffer.SafeRelease(); }`

Reply:
`NettyRlpStream` stream does this on `Dispose`. Probably we could have `PooledBufferNettyRlpStream` that manages allocation internally